File: solutions/c/resistor-color/1/resistor_color.c
#include "resistor_color.h"
#include <stdlib.h>
#include <stddef.h>
#include <string.h> // for memcpy

int color_code(resistor_band_t band_color) {
  return band_color;
}

resistor_band_t* colors() {
  resistor_band_t *colorList = malloc(10 * sizeof(resistor_band_t));
  memcpy(colorList, (resistor_band_t[]){ BLACK, BROWN, RED, ORANGE, YELLOW, GREEN, BLUE, VIOLET, GREY, WHITE }, 10 * sizeof(resistor_band_t));
  return colorList;
}

File: solutions/python/resistor-color/1/resistor_color.py
def color_code(color):
    try:
        return colors().index(color)
    except:
        return -1


def colors():
    return [
        "black",
        "brown",
        "red",
        "orange",
        "yellow",
        "green",
        "blue",
        "violet",
        "grey",
        "white",
    ]
